src/node_url.cc

```diff
@@ -676,10 +245,10 @@ bool BitAt(const uint8_t a[], const uint8_t i) {
 // Appends ch to str. If ch position in encode_set is set, the ch will
 // be percent-encoded then appended.
 void AppendOrEscape(std::string* str,
-                           const unsigned char ch,
-                           const uint8_t encode_set[]) {
+                    const unsigned char ch,
+                    const uint8_t encode_set[]) {
   if (BitAt(encode_set, ch))
-    *str += hex[ch];
+    *str += hex + (ch << 2);
```

Comment:
Let’s maybe add a comment here:

```suggestion
    *str += hex + ch * 4; // %00\0 has a length of 4
```

Reply:
done.

Reply:
@xadillax Why not just `* 4`? Using shifts for multiplication only is generally less readable

Reply:
I just think shift will be faster more or less.

Reply:
nope – this is an optimization that is *so* trivial that the compiler even applies it without any optimization levels enabled: https://godbolt.org/z/5d5xvaKPY

Reply:
done